### ISSUE TITLE
DuckDB sum requires CAST to INTEGER to work, or returns NULL

### ISSUE THREAD

**Issue by jschuur** [TablePlus/TablePlus#3466]
`SELECT SUM(value) from numbers` when `value` is already an INTEGER returns NULL in TablePlus, but works as expected in the DuckDB CLI.

Casting it explicitly to INTEGER again works in TablePlus

```sql
SELECT CAST(SUM(value) AS INTEGER) from numbers
```

Shouldn't this work without the CAST in TablePlus?

1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):

DuckDB 1.1.3

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):

Build 576

3. The steps to reproduce this issue:

```sql
-- create table with intege field
CREATE TABLE numbers (
    value INTEGER
);

-- add some values
INSERT INTO numbers (value) VALUES 
    (42),
    (123),
    (7);

-- data is there
SELECT * FROM numbers;

-- returns NULL in TablePlus, but 172 in the CLI
SELECT SUM(value) from numbers;

-- works as expected (172) in both the CLI and TablePlus
SELECT CAST(SUM(value) AS INTEGER) from numbers
```

**Comment by jschuur** [comment]
Also replicated this issue for a brand new `.db` file (`duckdb debug.db`), so it's not specific to the database I was working with.

**Comment by jschuur** [comment]
> Hi @jschuur I just fixed the issue on this build https://www.dropbox.com/scl/fi/rnzl329j9llknuoj0c0h1/TablePlus.dmg?rlkey=fykl1eshbl1ylnn48s5ic7rau&dl=0 Also support UUID and struct type in duckdb.

Amazing! Can confirm that it works without explicit casting on my end now too. Thanks for the super quick fix!

I did have similar unexpected NULLs turn up during aggregations or subqueries, but these always came from queries suggested from Claude AI, so I dismissed them as hallucinations.

I can't provide an example easily, so I'll start a new issue if I run into one again.
